packages/core/src/components/icon/_icon.scss

```diff
@@ -42,9 +42,6 @@ span.pt-icon {
 svg.pt-icon {
   // respect dimensions exactly
   flex: 0 0 auto;
-  // SVG and DOM elements don't align perfectly - this magic number seems to fix it.
-  // https://blog.prototypr.io/align-svg-icons-to-text-and-say-goodbye-to-font-icons-d44b3d7b26b4#6e0c
-  margin-top: -0.125em;
```

Comment:
`align-items` is much better at alignment than humans, so this shouldn't be necessary. keep an eye out for SVG icons mixed with text to ensure they're centered.